examples/sample_measurement/_array_utils.py

```diff
@@ -0,0 +1,50 @@
+"""Double2DArray Conversion Utilities."""
+
+from typing import Any
+
+from ni_measurement_plugin_sdk_service._internal.stubs.ni.protobuf.types import (
+    array_pb2,
+)
+
+try:
+    import numpy as np
+    from numpy.typing import NDArray
+
+    def double2darray_to_ndarray(double2darray: array_pb2.Double2DArray) -> NDArray[np.float64]:
+        """Convert Double2DArray to numpy NDArray."""
+        return np.array(double2darray.data, dtype=np.float64).reshape(
+            double2darray.rows, double2darray.columns
+        )
+
+    def ndarray_to_double2darray(ndarray: NDArray[np.float64]) -> array_pb2.Double2DArray:
+        """Convert numpy NDArray to Double2DArray."""
+        return array_pb2.Double2DArray(
+            data=ndarray.flatten().tolist(), rows=ndarray.shape[0], columns=ndarray.shape[1]
+        )
+
+except ImportError:
+    np = None
```

Comment:
Since you defined stub implementations of the functions, I don't think setting the numpy module to None is necessary.

Reply:
Updated in latest commit.